### ISSUE TITLE
Nextcloud crash since 0.11.0

### ISSUE THREAD

**Issue by Arno500** [gino0631/nextcloud-metadata#49]
Here is the log when accessing Files app:
### Renseignements techniques
Adresse distante : xx.xx.xx.xx
ID de la demande : dCkEF5RCOxJ3HBMb2qSb
Type : Exception
Code : 0
Message : Call to a member function getContentSecurityPolicyManager() on null
Fichier : /var/www/html/lib/private/AppFramework/Http/Dispatcher.php
Ligne : 109

Trace
```#0 /var/www/html/lib/private/AppFramework/App.php(126): OC\AppFramework\Http\Dispatcher->dispatch(Object(OCA\Files\Controller\ViewController), 'index')
#1 /var/www/html/lib/private/AppFramework/Routing/RouteActionHandler.php(47): OC\AppFramework\App::main('OCA\\Files\\Contr...', 'index', Object(OC\AppFramework\DependencyInjection\DIContainer), Array)
#2 [internal function]: OC\AppFramework\Routing\RouteActionHandler->__invoke(Array)
#3 /var/www/html/lib/private/Route/Router.php(297): call_user_func(Object(OC\AppFramework\Routing\RouteActionHandler), Array)
#4 /var/www/html/lib/base.php(997): OC\Route\Router->match('/apps/files/')
#5 /var/www/html/index.php(42): OC::handleRequest()
#6 {main}
```
The problem goes away when disabling the plugin (and it worked perfectly in 0.10.0)

Note: I use NC 17.0.2

**Comment by Uninen** [comment]
I bumped into this same problem, also when upgrading to 0.11.0 from 0.10.0. I have a slightly different error message:

```
Error: Call to a member function getContentSecurityPolicyManager() on null
```

and 

```
Undefined variable: server at /var/snap/nextcloud/18204/nextcloud/extra-apps/metadata/lib/AppInfo/Application.php#31
```
I'm running 16.0.7 (Snap)

**Comment by fa-ribeiro** [comment]
same here
NC 17.0.2
error after update metadata from  0.10.0 to  0.11.0:
`Exception: Call to a member function getContentSecurityPolicyManager() on null`
and 
`Undefined variable: server at /var/www/nextcloud/apps/metadata/lib/AppInfo/Application.php#31`

**Comment by nursoda** [comment]
Same in NC 18. I see an internal server error instead of the NC interface if insstalled, no NGINX err log.

**Comment by jgerken** [comment]
Have the same version and error message as fa-ribeiro. Didn‘t try his fix.

**Comment by gerroon** [comment]
Same problem,. crashes Nextcloud. I had to manually delete itm from the server.

**Comment by xsus95** [comment]
How can i manually delete or deactivate it? Via occ? :(

**Comment by gerroon** [comment]
@xsus95, I just opened the apps folder in the terminal and removed tte metadata folder literally.

**Comment by CSchank** [comment]
I am also having issues, with Nextcloud 17.0.2 and metadata 0.11.0. Turning on `occ log:watch` produces the following:

```
 Undefined variable: server at /usr/share/nginx/nextcloud/apps/metadata/lib/AppInfo/Application.php#31                                                                                      2020-01-21T04:48:16+00:00 

  Error    index              Exception: Call to a member function getContentSecurityPolicyManager() on null at lib/private/AppFramework/Http/Dispatcher.php line 109                                                    2020-01-21T04:48:16+00:00 
                                                                                                                                                                                                                                                   
                              0. lib/private/AppFramework/App.php line 126                                                                                                                                                                         
                                 OC\AppFramework\Http\Dispatcher->dispatch(OCA\Files\Controller\ViewController {}, "index")                                                                                                                        
                              1. lib/private/AppFramework/Routing/RouteActionHandler.php line 47                                                                                                                                                   
                                 OC\AppFramework\App::main("OCA\\Files\\Controller\\ViewController", "index", OC\AppFramework\DependencyInjection\DIContainer {}, {_route:"files.view.index"})                                                     
                              2. <<closure>>                                                                                                                                                                                                       
                                 OC\AppFramework\Routing\RouteActionHandler->__invoke({_route:"files.view.index"})                                                                                                                                 
                              3. lib/private/Route/Router.php line 297                                                                                                                                                                             
                                 call_user_func(OC\AppFramework\Routing\RouteActionHandler {}, {_route:"files.view.index"})                                                                                                                        
                              4. lib/base.php line 997                                                                                                                                                                                             
                                 OC\Route\Router->match("\/apps\/files\/")                                                                                                                                                                         
                              5. index.php line 42                                                                                                                                                                                                 
                                 OC::handleRequest(                                                                                                                                                                                                
                                                                                                                                                                                                                                                   
                                 )                                                 
```

Disabling the app via `occ` makes the server work again normally, re-enabling causes this every time.

**Comment by tht** [comment]
Same issue here.
It is possible to remove it from the server using the web ui as the settings are still accessible. Jump directly to the URL: https://nextcloud-url/settings/apps/enabled and disable "Metadata" there.

**Comment by pipuvic** [comment]
> @fa-ribeiro's fix worked for me.

for me too! Thanks!

**Comment by probastaller** [comment]
Same here using nginx.
Had to remove the entire "metadata" folder from apps.

**Comment by idef1x** [comment]
@fa-ribeiro's fix worked for me too :)

**Comment by mjanssens** [comment]
Or command line:
`sudo -u www-data php /var/www/nextcloud/occ app:disable metadata`
Of course while adjusting to the user your webserver runs with and the path where occ is located.

**Comment by jacotec** [comment]
I confirm also @fa-ribeiro's fix! Thank you for sharing this! :-)

**Comment by vanoc** [comment]
Same. Disabled Metadata.

**Comment by gino0631** [comment]
Sorry for this :(

**Comment by gino0631** [comment]
Fixed version 0.11.1 is already uploaded. Sorry once again for the problems :(

**Comment by Tie-fighter** [comment]
Thank you for the quick fix!
